### PR TITLE
some coverity fixes

### PR DESCRIPTION
**Pull request by Goober5000**
Check the return value of `eval_wing()`, and tweak the checks for `closeup_pos` in `draw_model_rotating()`.  This addresses a few Coverity defects.

EDIT: Also address two false positives and refine the usage of `sexp_get_variable_index()`.

### CODE REVIEW

File: code/io/timer.h [hunk 1]
```diff
@@ -63,6 +63,7 @@ class TIMESTAMP : public util::ID<timestamp_tag, int, -1>
 
 	inline bool isNever() const { return m_val == 0; }
 	inline bool isImmediate() const { return m_val == 1; }
+	inline bool isFinite() const { return isValid() && !isNever(); }
 };
```
Comment (JohnAFernandez) [inline code/io/timer.h:67]: Honestly, these timestamp changes should be separate.

Comment (Goober5000) [reply code/io/timer.h:67]: Ok

Comment (Goober5000) [reply code/io/timer.h]: I've split them off into PR #6278

**Review by wookieejedi — APPROVED** [review]
Overall looks good to me but happy to have another set of eyes if anyone wants to.

**Pull request merged**